### PR TITLE
1072/same email add user

### PR DESCRIPTION
**Pull request by P-Saha**
**Description of changes**

**Type of Change**

- [x] Bug
- [ ] Styling
- [ ] New Feature

**Checklist before requesting review:**

- [ ] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [x] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

File: components/pages/submission-system/modals/addUser/index.tsx [hunk 1]
```diff
@@ -81,16 +83,36 @@ const AddUserModal = ({
...
+    return false;
+  };
+
   // validate each individual form and fire onSubmit for each
   const submitForm = async () => {
     const allForms = Object.keys(formSubscriptions).map(async key => {
       const form = formSubscriptions[key];
       return form.validateForm(form.data);
     });
+    const allFormsDuplicateEmail = Object.keys(formSubscriptions) // True if a duplicate Email exists
```
Comment (hlminh2000) [inline components/pages/submission-system/modals/addUser/index.tsx]: `Object.values` work better if the values are the only interest. Also `.some` reads better, so, this whole block can be reduced to:
```
Object.values(formSubscriptions)
    .some(formHasDuplicateEmail)
```

where `formHasDuplicateEmail` is `duplicateEmailValidate`, just reads better :P

File: components/pages/submission-system/modals/addUser/index.tsx [hunk 2]
```diff
@@ -81,16 +83,36 @@ const AddUserModal = ({
     setIsLastSectionTouched(lastSection.touched);
   };
 
+  // sets Error and returns true if an error exists
+  const duplicateEmailValidate = form => {
+    const programEmails = Object.keys(formSubscriptions)
+      .map(key => formSubscriptions[key].data.email)
+      .concat(Object.keys(users).map(key => users[key].email));
```
Comment (hlminh2000) [inline components/pages/submission-system/modals/addUser/index.tsx]: isn't users an array? can just do `users.map(({email}) => email)`

File: components/pages/submission-system/modals/addUser/index.tsx [hunk 3]
```diff
@@ -81,16 +83,36 @@ const AddUserModal = ({
...
+    if (new Set(programEmails).size !== programEmails.length) {
```
Comment (hlminh2000) [inline components/pages/submission-system/modals/addUser/index.tsx]: functionally the same, but `uniq(programEmails).length !== programEmails.length` reads better :) (https://lodash.com/docs/4.17.15#uniq)

File: components/pages/submission-system/modals/addUser/index.tsx [hunk 4]
```diff
@@ -68,10 +70,10 @@ const AddUserModal = ({
   // check for errors
   const errorCheck = () => {
     const formSubKeys = Object.keys(formSubscriptions);
-    const validity = formSubKeys
+    const invalidity = formSubKeys
```
Comment (hlminh2000) [inline components/pages/submission-system/modals/addUser/index.tsx]: might as well rewrite this to `Object.values(formSubscriptions).some(form => form.hasErrors)`

File: components/pages/submission-system/modals/addUser/index.tsx (hunk 1 above)
Comment (hlminh2000) [reply components/pages/submission-system/modals/addUser/index.tsx]: `allFormsDuplicateEmail` doesn't sound like a boolean... `someFormHasDuplicateEmail` is more expressive :P it's already a long name, might as well reflect what it is.

File: components/pages/submission-system/modals/addUser/index.tsx [hunk 5]
```diff
@@ -8,9 +8,9 @@ import { UserSection, UserSectionProps } from '../styledComponents';
 import useFormHook from 'global/hooks/useFormHook';
 import { UserModel, userSchema } from '../common';
 import uniqueId from 'lodash/uniqueId';
-import isEmpty from 'lodash/isEmpty';
+import { firstName } from 'global/utils/form/validations';
 
-const AddUser = ({ id, formSubscriptions, removeSection, onUpdate, showDelete }) => {
+const AddUser = ({ id, formSubscriptions, removeSection, onUpdate, showDelete, users }) => {
```
Comment (hlminh2000) [inline components/pages/submission-system/modals/addUser/index.tsx]: is `users` used?

File: components/pages/submission-system/modals/addUser/index.tsx [hunk 6]
```diff
@@ -151,6 +173,7 @@ const AddUserModal = ({
               lastSectionTouchCheck();
             }}
             showDelete={formIds.length > 1}
+            users={users}
```
Comment (hlminh2000) [inline components/pages/submission-system/modals/addUser/index.tsx]: does `AddUser` actually uses `users`?

File: components/pages/submission-system/modals/addUser/index.tsx [hunk 7]
```diff
@@ -46,9 +46,11 @@ const AddSection = styled(Button)`
 const AddUserModal = ({
   onSubmit,
   dismissModal,
+  users,
 }: {
   onSubmit: (data: typeof UserModel[]) => any | void;
   dismissModal: () => any | void;
+  users: [];
```
Comment (hlminh2000) [inline components/pages/submission-system/modals/addUser/index.tsx]: `[]` is a tuple type. This should be `Array<typeof UserModel>`

File: components/pages/submission-system/modals/addUser/index.stories.tsx [hunk 8]
```diff
@@ -7,6 +7,7 @@ const AddUserModalStory = storiesOf(`${__dirname}`, module).add('Basic', () => (
   <AddUserModal
     onSubmit={data => action('Submit data')(data)}
     dismissModal={action('dismissModal')}
+    users={null}
```
Comment (hlminh2000) [inline components/pages/submission-system/modals/addUser/index.stories.tsx]: probably `[]` is a better empty value. or maybe give it an actual empty sample?

File: components/pages/submission-system/modals/addUser/index.tsx (hunk 1 above)
Comment (P-Saha) [reply components/pages/submission-system/modals/addUser/index.tsx]: When using .some, the higher order function stops at the first fail, when it is preferred to have all of the functions run for every value. I will however keep the rest in mind!

Comment (hlminh2000) [reply components/pages/submission-system/modals/addUser/index.tsx]: gotcha. In that case, it's better to extract the side effect outside and trigger it here. Having a side effect inside a map callback is trippy...

**Pull request merged**